src/main/ank/2.2-limits-and-colimits.md

```diff
@@ -0,0 +1,208 @@
+```Haskell
+p' = p . m
+q' = q . m
+```
+```kotlin
+val `p'` = p compose m
+val `q'` = q compose m
+```
+................
+```Haskell
+contramap :: (c' -> c) -> (c -> LimD) -> (c' -> LimD)
+contramap f u = u . f
+```
+```kotlin:ank:silent
+import arrow.core.compose
+object LimD
+
+fun <A, B> contramap(f: (A) -> B, u: (B) -> LimD): ((A) -> LimD) = u compose f
+```
+................
+```Haskell
+f :: a -> b
+g :: a -> b
+```
+```kotlin
+val f: (A) -> B
+val g: (A) -> B
+```
+................
+```Haskell
+p :: c -> a
+q :: c -> b
+```
+```kotlin
+val p: (C) -> A
+val q: (C) -> B
+```
+................
+```Haskell
+q = f . p
+q = g . p
+```
+```kotlin
+val q = f compose p
+val q = g compose p
+```
+................
+```Haskell
+f . p = g . p
+```
+```kotlin
+f compose p == g compose p
+```
+................
+```Haskell
+f (x, y) = 2 * y + x
+g (x, y) = y - x
+```
+```kotlin:ank:playground
+val f = { x: Int, y: Int -> 2 * y + x }
+val g = { x: Int, y: Int -> y - x }
+```
+................
+```Haskell
+p t = (t, (-2) * t)
+```
+```kotlin:ank:playground
+import arrow.core.toT
+
+val p = { t: Int -> t toT (-2) * t }
+```
+................
+```Haskell
+f . p' = g . p'
+```
+```kotlin
+f compose `p'` == g compose `p'`
+```
+................
+```Haskell
+p'() = (0, 0)
+```
+```kotlin:ank:playground
+import arrow.core.toT
+
+fun `p'`() = 0 toT 0
+```
+................
+```Haskell
+h () = 0
+```
+```kotlin:ank:plaground
+fun h() = 0
+```
+................
+```Haskell
+f :: a -> b
+g :: c -> b
+```
+```kotlin
+val f: (A) -> B
+val g: (C) -> B
+```
+................
+```Haskell
+p :: d -> a
+q :: d -> c
+r :: d -> b
+```
+```kotlin
+val p: (D) -> A
+val q: (D) -> C
+val r: (D) -> B
+```
+................
+```Haskell
+g . q = f . q
+```
+```kotlin
+g compose q == f compose q
+```
+................
+```Haskell
+f x = 1.23
+```
+```kotlin
+val f: (A) -> Double = { _: A -> 1.23 }
+```
+................
+```Haskell
+twice f x = f (f x)
+```
+```kotlin
+fun <A> twice(): ((((A) -> A), A) -> A) =
```

Comment:
It doesn't need to hint the types here and in the lambda parameters. It also has an extra set of ( )